javascript/ql/src/Security/CWE-020/UselessRegExpCharacterEscape.qhelp

```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE qhelp PUBLIC
+"-//Semmle//qhelp//EN"
+"qhelp.dtd">
+<qhelp>
+
+	<overview>
+		<p>
+
+			Prepending a single backslash to a non-special character in a
+			string literal, template literal, or regular expression literal
+			does not have any effect. This can make the character appear to be
+			escaped during a cursory look at the source code even though the
+			character is not escaped at runtime.
+
+			To make matters more complex, regular expression literals
+			have a slightly different set of special characters than string
+			literals and template literals.
+
+		</p>
+
+	</overview>
+
+	<recommendation>
+		<p>
+
+			Ensure that the right amount of backslashes is used when
+			escaping characters in strings, template literals and regular
+			expressions.
+
+			Pay special attention to the number of backslashes when
+			rewriting a regular expression as a string.
+
+		</p>
+	</recommendation>
+
+	<example>
+
+		<p>
+
+			The following example code checks that a string is
+			<code>"my-marker"</code>, possibly surrounded by white space.
+
+		</p>
+
+		<sample src="examples/UselessRegExpCharacterEscape_bad_1.js"/>
+
+		<p>
+
+			However, the check does not work properly for white space
+			as the two <code>\s</code> occurrences are semantically equivalent to
+			just <code>s</code>, meaning that the check will succeed for strings
+			like <code>"smy-markers"</code> instead of <code>" my-marker
```

Comment:
Just double-checking that the white spaces before and after the second expression are deliberate here ;-)